tests/integration/components/polaris-button-test.js

```diff
@@ -1,639 +1,333 @@
-  );
-
-  // Button with aria expanded.
-  this.render(hbs`{{polaris-button ariaExpanded=ariaExpanded}}`);
-
-  buttons = findAll('button[type="button"].Polaris-Button');
-  assert.equal(buttons.length, 1, 'button with aria expanded - renders');
-
-  button = buttons[0];
-  assert.notOk(
-    button.attributes['aria-expanded'],
-    'button with aria expanded undefined - does not set aria-expanded'
-  );
-
-  this.set('ariaExpanded', false);
-  assert.equal(
-    button.attributes['aria-expanded'].value,
-    'false',
-    'button with aria expanded set - sets aria-expanded'
-  );
-});
+module('Integration | Component | polaris-button', function(hooks) {
+  setupRenderingTest(hooks);
 
-test('it supports an icon source value for `icon`', function(assert) {
-  this.render(hbs`{{polaris-button icon=icon text=text}}`);
-
-  let button = find('button.Polaris-Button');
-  let icon = find(buttonIconWrapperSelector, button);
-  assert.notOk(icon, 'button without icon - no icon rendered');
-
-  this.set('icon', 'cancel');
-
-  icon = find(buttonIconWrapperSelector, button);
-  let iconSvg = find('span.Polaris-Icon > svg.Polaris-Icon__Svg', icon);
-  assert.ok(icon, 'button with icon - renders icon');
-  assert.equal(
-    iconSvg.dataset.iconSource,
-    'polaris/cancel',
-    'button with icon - renders correct icon SVG'
-  );
-  assert
-    .dom(button)
-    .hasClass(
-      'Polaris-Button--iconOnly',
-      'button with icon - without text, has icon-only class'
-    );
-
-  this.set('text', 'Some text');
-  assert
-    .dom(button)
-    .hasNoClass(
-      'Polaris-Button--iconOnly',
-      'button with icon - with text, does not have icon-only class'
-    );
-});
+  hooks.beforeEach(function() {
+    this.owner.register('component:svg-jar', MockSvgJarComponent);
+  });
 
-test('it supports a component value for `icon`', function(assert) {
-  this.render(hbs`
-    {{polaris-button
-      text=""
-      icon=(component "polaris-avatar"
-        name="Tom"
-      )
-    }}
-  `);
-
-  let buttonIconComponentSelector = buildNestedSelector(
-    buttonIconWrapperSelector,
-    '.Polaris-Avatar'
-  );
-
-  assert
-    .dom(buttonIconComponentSelector)
-    .exists('component renders inside the icon wrapper');
-  assert
-    .dom(buttonIconSelector)
-    .doesNotExist('icon is not rendered inside the icon wrapper');
-});
+  module('<Button />', function() {
+    module('url', function() {
+      hooks.beforeEach(function() {
+        this.set('mockUrl', 'http://google.com');
+      });
+
+      test('renders a link when present', async function(assert) {
+        await render(hbs`{{polaris-button url=mockUrl}}`);
+        assert.dom('a[data-polaris-unstyled]').exists();
+      });
+
+      test('gets passed into the link', async function(assert) {
+        await render(hbs`{{polaris-button url=mockUrl}}`);
+        assert
+          .dom('a[data-polaris-unstyled]')
+          .hasAttribute('href', this.get('mockUrl'));
```

Comment:
no biggie (or need to update), but we should really start to use ES6 getters

Reply:
Do ES6 getters work with older versions of Ember (e.g. 2.18, which we still support)? Or do they cause any potential issues with apps that use older versions of Ember?